src/yaml-tools.js

```diff
@@ -111,11 +123,30 @@ yamlLibrary = {
   // doc: The SAM.YML parsed document
   // schema: the JSON schema to compare the doc to
   checkDocAgainstSchema: function (doc, schema, failed, warning, info) {
-    this.doc = doc;
+    this.doc = this.docs[doc];
     var keys = [];
 
     this.recurseIntoArray(schema, keys, null, failed, warning, info);
+  },
+
+  loadDocData: function (docs) {
+    let docsArray = cleanDocsData(docs).split('---');
+
+    for (let i = 0; i < docsArray.length; i++) {
+      let allComments = true;
+      let yamlDoc = yaml.load(docsArray[i]);
+      this.docs[i] = yamlDoc == null || yamlDoc == undefined ? {} : yamlDoc;
```

Comment:
You explained how this is working to me, so I think a comment here would be very useful, so others know how this `{}` will ultimately cause the validation to fail.